pkg/cmd/roachtest/tests/sstable_corruption.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+)
+
+func runSstableCorruption(ctx context.Context, t test.Test, c cluster.Cluster) {
+	crdbNodes := c.Range(1, c.Spec().NodeCount)
+	workloadNode := c.Node(1)
+	const corruptNode = 3
+
+	t.Status("installing cockroach")
+	c.Put(ctx, t.Cockroach(), "./cockroach", crdbNodes)
+	c.Start(ctx, crdbNodes)
+
+	// We don't really need tpcc, we just need a good amount of data. Enough
+	// to have multiple ranges, and some sstables with only table keys.
+	t.Status("importing tpcc fixture")
+	c.Run(ctx, workloadNode,
+		"./cockroach workload fixtures import tpcc --warehouses=100 --fks=false --checks=false")
+
+	m := c.NewMonitor(ctx, crdbNodes)
+	signalChan := make(chan bool)
+
+	m.Go(func(ctx context.Context) error {
+		// Wait for a signal from the other goroutine below before running the
+		// workload. This will only be called after the cluster has been restarted
+		// with corrupt sstables on one node.
+		select {
+		case proceed := <-signalChan:
+			if !proceed {
+				return nil
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+		_ = c.RunE(ctx, workloadNode, "./cockroach workload run tpcc --warehouses=100 "+
+			fmt.Sprintf("--tolerate-errors --duration=%s", 5*time.Minute))
+		return nil
+	})
+
+	m.Go(func(ctx context.Context) error {
+		m.ExpectDeaths(3)
```

Comment:
I think this would end up easier if you used locally-scoped monitors:

```go
{
  c.Start(/* ... */)
  m := c.NewMonitor(ctx, crdbNodes)
  m.Go(/* import */)
  m.Wait()
  c.Stop()
  // now corrupt the sst
}

{
  err := c.StartE() // if error here, test is done
  m := NewMonitor()
  m.Go(/* workload */)
  require.IsError(t, m.WaitE())
}

{
  // WaitE returned an error, so either a node crashed and/or workload
  // returned an error (it certainly did if a node crashed), make sure that a node crashed.
  m := Monitor()
  require.IsError(t, m.WaitE)
}
```